tests/PaymentPart/Output/FpdfOutput/FpdfOutputTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Sprain\Tests\SwissQrBill\PaymentPart\Output\FpdfOutput;
+
+use PHPUnit\Framework\TestCase;
+use Sprain\SwissQrBill\PaymentPart\Output\FpdfOutput\FpdfOutput;
+use Sprain\SwissQrBill\PaymentPart\Output\FpdfOutput\Template\QrBillFooter;
+use Sprain\SwissQrBill\QrBill;
+use Sprain\SwissQrBill\QrCode\QrCode;
+use Sprain\Tests\SwissQrBill\TestQrBillCreatorTrait;
+
+class FpdfOutputTest extends TestCase
+{
+    use TestQrBillCreatorTrait;
+
+    /**
+     * @dataProvider validQrBillsProvider
+     * @param string $name
+     * @param QrBill $qrBill
+     */
+    public function testValidQrBills(string $name, QrBill $qrBill)
+    {
+        $variations = [
+            [
+                'printable' => false,
+                'format' => QrCode::FILE_FORMAT_PNG,
+                'file' => dirname(dirname(dirname(__DIR__))) . '/TestData/FpdfOutput/' . $name . '.pdf'
+            ],
+            [
+                'printable' => true,
+                'format' => QrCode::FILE_FORMAT_PNG,
+                'file' => dirname(dirname(dirname(__DIR__))) . '/TestData/FpdfOutput/' . $name . '.print.pdf'
+            ]
+
+            // Note: Testing the exact output with a png qr code is not possible, as the png contents are
+            // not always exactly the same on each server configuration.
```

Comment:
I am surprised that tests actually do work with png :) 
So this comment can be removed here and I will look into this in other tests.

Reply:
This is because I changed a bit your code in placeholder and add marks.png in asset directory.
```
public function getFile(?string $ext = 'svg'): ?string
    {
        if ($ext === 'png') {
            $this->file = str_replace('.svg', '.png', $this->file);
        }
        return $this->file;
    }
```

Reply:
Ok, that's quite a hack ;) 
But still, the tests pass, and that's what surprises me.